apps/session-manager/src/controllers/__tests__/zendesk.spec.ts

```diff
@@ -0,0 +1,244 @@
+      TE.left(new Error("ERROR while generating JWT support token")),
+    );
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error("ERROR while generating JWT support token"),
+        ),
+      ),
+    )();
+  });
+
+  it("should return a IResponseErrorInternal if getProfile returns an error", async () => {
+    mockGetProfile.mockReturnValue(TE.left(Error("")));
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error(
+            "Error retrieving a user profile with validated email address | ",
+          ),
+        ),
+      ),
+    )();
+  });
+
+  it("should return a IResponseErrorInternal if getProfile promise gets resolved with a IResponseErrorInternal", async () => {
+    mockGetProfile.mockReturnValue(
+      TE.right(ResponseErrorInternal("Any Error")),
+    );
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error(
+            "Error retrieving a user profile with validated email address | Error retrieving user profile | Internal server error: Any Error",
+          ),
+        ),
+      ),
+    )();
+  });
+
+  it("should return a IResponseErrorInternal if getProfile promise gets resolved with a IResponseErrorTooManyRequests", async () => {
+    mockGetProfile.mockReturnValue(
+      TE.right(ResponseErrorTooManyRequests("Rate limit triggered")),
+    );
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error(
+            "Error retrieving a user profile with validated email address | Error retrieving user profile | Too many requests: Rate limit triggered",
+          ),
+        ),
+      ),
+    )();
+  });
+
+  it("should return a IResponseErrorInternal if getProfile promise gets resolved with a IResponseErrorNotFound", async () => {
+    mockGetProfile.mockReturnValue(
+      TE.right(ResponseErrorNotFound("User not found", "Cannot find user")),
+    );
+
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.mapLeft((response) =>
+        expect(response).toEqual(
+          Error(
+            "Error retrieving a user profile with validated email address | Error retrieving user profile | User not found: Cannot find user",
+          ),
+        ),
+      ),
+    )();
+  });
+
+  it("should return a IResponseErrorValidation if request's user cannot be validated", async () => {
+    await pipe(
+      mockedDependencies,
+      ZendeskController.getZendeskSupportToken,
+      TE.map((response) =>
+        expect(response.kind).toEqual("IResponseErrorValidation"),
+      ),
+    )();
+  });
```

Comment:
This check is not performed.

Reply:
what do you mean here? based on the implementation, a `TE.right` is expected with an `IResponseErrorValidation`